### PR TITLE
python3.pkgs.paramiko: 3.4.1 -> 3.5.0

### PR DESCRIPTION
**Pull request by Conni2461**
## Description of changes

Changelog: https://www.paramiko.org/changelog.html#3.5.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/paramiko/default.nix
Comment (momeemt) [inline pkgs/development/python-modules/paramiko/default.nix:19]: Could you remove the `rec` keyword? (ref: https://nix.dev/guides/best-practices#recursive-attribute-set-rec)
You can define following to remove it.

```suggestion
let
  pname = "paramiko";
  version = "3.5.0";
in
buildPythonPackage {
```

File: pkgs/development/python-modules/paramiko/default.nix
Comment (momeemt) [inline pkgs/development/python-modules/paramiko/default.nix:21]:
```suggestion
```

File: pkgs/development/python-modules/paramiko/default.nix
Comment (momeemt) [inline pkgs/development/python-modules/paramiko/default.nix:20]:
```suggestion
  inherit pname version;
```

File: pkgs/development/python-modules/paramiko/default.nix
Comment (momeemt) [inline pkgs/development/python-modules/paramiko/default.nix:28]: Could you remove `with lib;` in line 68? (ref: https://github.com/NixOS/nixpkgs/issues/208242)
You can use `lib.licenses.lgpl21Plus` instead of `licenses.lgpl21Plus`.

Comment (momeemt) [inline pkgs/development/python-modules/paramiko/default.nix:28]: Are you interested in becoming a maintainer for this package? No one is currently maintaining it. If you're not interested, sorry to bother you, could you add me (momeemt) to the maintainer list (line 79) as follows `maintainers = with lib.licenses.maintainers [ momeemt ];`?

**Review by momeemt — CHANGES_REQUESTED** [review]
Thanks for your contribution. I left the comment about `rec`, `with lib;` and maintainer list.

**Comment by dasJ** [comment]
Please don't bikeshed this update PR by requesting various refactorings. Those should be done in a different PR.

**Review by dasJ — APPROVED** [review]
Changes LGTM

File: pkgs/development/python-modules/paramiko/default.nix
Comment (Conni2461) [reply pkgs/development/python-modules/paramiko/default.nix:28]: we are fine with maintaining it, doing this kinda already. i've also done the last bump

**Comment by momeemt** [comment]
> Please don't bikeshed this update PR by requesting various refactorings. Those should be done in a different PR.

Sorry, I misunderstood that  it should be improved as much as possible before being merged. I'll be sure to do what you said the next time I review PRs.

**Review by momeemt — APPROVED** [review]
LGTM

**Comment by mweinelt** [comment]
Please reference a changelog in the commit message.

**Pull request merged**
